File: frontend/extract_resource_hub.js
const { execSync } = require('child_process');
const fs = require('fs');

try {
  const content = execSync('git show 64d8a4c:frontend/src/pages/DashboardPage.js', { encoding: 'utf8' });
  const lines = content.split('\n');
  
  let startIdx = -1;
  let endIdx = -1;
  for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('function ResourceHub')) {
      startIdx = i;
      break;
    }
  }

  if (startIdx !== -1) {
    let braceCount = 0;
    let started = false;
    for (let i = startIdx; i < lines.length; i++) {
      const line = lines[i];
      if (line.includes('{')) {
        started = true;
        braceCount += (line.match(/{/g) || []).length;
      }
      if (line.includes('}')) {
        braceCount -= (line.match(/}/g) || []).length;
      }
      if (started && braceCount === 0) {
        endIdx = i;
        break;
      }
    }
    const result = lines.slice(startIdx, endIdx + 1).join('\n');
    fs.writeFileSync('frontend/extracted_resource_hub.js', result, 'utf8');
    console.log('ResourceHub extracted to frontend/extracted_resource_hub.js');
  } else {
    console.log('ResourceHub not found');
  }
} catch (err) {
  console.error(err);
}

File: frontend/extract_resource_hub.py
import subprocess

def get_file_content_at_commit(commit, filepath):
    cmd = ["git", "show", f"{commit}:{filepath}"]
    res = subprocess.run(cmd, capture_output=True, text=True)
    return res.stdout

# Get DashboardPage.js at first commit
content = get_file_content_at_commit("64d8a4c", "frontend/src/pages/DashboardPage.js")
lines = content.splitlines()

# Find ResourceHub
start_idx = -1
end_idx = -1
for i, line in enumerate(lines):
    if "function ResourceHub" in line:
        start_idx = i
        break

if start_idx != -1:
    # Find matching brace
    brace_count = 0
    started = False
    for i in range(start_idx, len(lines)):
        line = lines[i]
        if "{" in line:
            started = True
            brace_count += line.count("{")
        if "}" in line:
            brace_count -= line.count("}")
        if started and brace_count == 0:
            end_idx = i
            break

    print(f"ResourceHub found at lines {start_idx} to {end_idx}:")
    print("\n".join(lines[start_idx:end_idx+1]))
else:
    print("ResourceHub not found")
